core/app/models/spree/credit_card.rb

```diff
@@ -5,7 +5,6 @@ class CreditCard < Spree::Base
     # to migrate safely from older Spree versions that din't provide safe deletion for CCs
     # we need to ensure that we have a connection to the DB and that the `deleted_at` column exists
     if !ENV['SPREE_DISABLE_DB_CONNECTION'] &&
-        connected? &&
```

Comment:
@damianlegawiec could you please take a look on this line? before remove it breaks specs